### ISSUE TITLE
malformed test chunk results in unfriendly error

### ISSUE THREAD

**Issue by cdent** [cdent/gabbi#63]
if you create (note the missing `-` before `name`):

```yaml
tests:
    name: foo
    url: /
    stauts: 404
```

You'll get a traceback:

```
  File "/Users/cdent/src/gabbi/gabbi/driver.py", line 120, in test_update
    for key, val in six.iteritems(new_dict):
  File "/Library/Python/2.7/site-packages/six.py", line 576, in iteritems
    return iter(d.iteritems(**kw))
AttributeError: 'str' object has no attribute 'iteritems'
```

We can do better than that.

**Comment by cdent** [comment]
see #64

**Comment by cdent** [comment]
closed by #64
